google-cloud-bigtable/acceptance/bigtable/table/mutate_row_test.rb

```diff
@@ -74,6 +74,89 @@
     row.cells[family].select{|v| v.qualifier == qualifier}.length.must_equal 1
   end
 
+  it "raises when set_cell with incorrect Time timestamp" do
+    postfix = random_str
+    row_key = "setcell-#{postfix}"
+    qualifier = "mutate-row-#{postfix}"
+
+    table.granularity.must_equal :MILLIS
+
+    # Set cell
+    entry = table.new_mutation_entry(row_key)
+    err = expect do
+      entry.set_cell(
+        family, qualifier, "mutatetest value #{postfix}", timestamp: Time.now
+      )
+    end.must_raise Google::Protobuf::TypeError
+    err.message.must_match "Expected number type for integral field 'timestamp_micros' (given Time)."
+  end
+
+  it "raises when set_cell with incorrect microsecond integer timestamp" do
+    postfix = random_str
+    row_key = "setcell-#{postfix}"
+    qualifier = "mutate-row-#{postfix}"
+    t = Time.now
+    timestamp_micros = t.to_i * 1000000 + t.usec
```

Comment:
We have a 1 in 1000 chance of `` ending in `000` and this test failing. Perhaps add a check for this here?

```suggestion
    timestamp_micros = (Time.now.to_f * 1000000).round
    timestamp_micros += 1 if (timestamp_micros % 10).zero?
```